packages/core/src/messagingProtocols/mls/MLSService/stores/subconversationGroupIdStore/subconversationGroupIdStore.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {QualifiedId} from '@wireapp/api-client/lib/user';
+
+const storageKey = 'subconversationGroupIdStore';
+
+function generateSubconversationId(parentConversation: QualifiedId, subconversation: string): string {
+  return `${parentConversation.id}@${parentConversation.domain}:${subconversation}`;
+}
+
+function parseSubconversationId(subconversationId: string): {parentConversation: QualifiedId; subconversation: string} {
+  const [parentConversationId, subconversation] = subconversationId.split(':');
+  const [id, domain] = parentConversationId.split('@');
+  return {parentConversation: {domain, id}, subconversation};
+}
+
+const getCurrentMap = (): Map<string, string> => {
+  const storedEntry = localStorage.getItem(storageKey);
+  return storedEntry ? new Map<string, string>(JSON.parse(storedEntry)) : new Map<string, string>();
+};
+
+const addItemToMap = (subconversationId: string, subgroupId: string) => {
+  const currentMap = getCurrentMap();
+  currentMap.set(subconversationId, subgroupId);
+  localStorage.setItem(storageKey, JSON.stringify(Array.from(currentMap.entries())));
+};
+
+const removeItemFromMap = (subconversationId: string) => {
+  const currentMap = getCurrentMap();
+  currentMap.delete(subconversationId);
+  localStorage.setItem(storageKey, JSON.stringify(Array.from(currentMap.entries())));
+};
+
+function storeGroupId(parentConversation: QualifiedId, subconversation: string, subgroupId: string): void {
+  const subconversationId = generateSubconversationId(parentConversation, subconversation);
+  addItemToMap(subconversationId, subgroupId);
+}
+
+function getGroupId(parentConversation: QualifiedId, subconversation: string): string | undefined {
```

Comment:
This is what we were doing before see two deleted files in this PR (they should be renamed not recreated - thanks git 😓). What I've added is storing the map in localStorage + removal option, so we can keep track of "known" subconversations -> we can access list of subconversations after client crash and leave them.

Reply:
maybe we should rename this module adding the idea that those subconv are only temporary. 

I fear if we make it too general purposes it's going to be used in the future for subconv that should stay long lived and then it's going to backfire by deleting the conversation on load. 

Maybe `TemporarySubconversationGroupIdStore` or something like this. WDYT?

Reply:
Yeah, good idea, even though we're using `localStorage` they're not eternal. Good call 👌

Reply:
Resolved on 1:1 chat: We've ended up keeping the storage as it is, we changed `getAllGroupIds` to `getAllGroupIdsBySubconversationId` to be sure only subconversations of specified id (eg. `"conference"`) are returned (and cleared on app reload).

With this implementation store has only one responsibility: persistent storage of groupIds. Consumer is able to remove/add entries whenever it want.